app/renderer/actions/Inspector.js

```diff
@@ -431,26 +431,39 @@ export function searchForElement (strategy, selector) {
   };
 }
 
-
 export function resetSearchForPORootElement () {
   return (dispatch) => {
-    dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS});
+    dispatch({type: SEARCHING_FOR_ROOT_ELEMENT});
   };
 }
 
-export function searchForPORootElement (strategy, selector, po) {
+export function searchForPORootElements (treeData) {
+  const strategyMap = {
+    'accessid': 'accessibility id',
+    'uiautomator': '-android uiautomator',
+    'classchain': '-ios class chain',
+  };
   return async (dispatch, getState) => {
     try {
-      const callAction = callClientMethod({strategy, selector, fetchArray: true});
-      let {elements, variableName} = await callAction(dispatch, getState);
-      const findAction = findAndAssign(strategy, selector, variableName, true);
-      findAction(dispatch, getState);
-      elements = elements.map((el) => el.id);
-      if (elements.length > 0) {
-        elements.push(po.root);
-        dispatch({type: SEARCHING_FOR_ROOT_ELEMENTS_COMPLETED, elements});
+      dispatch({type: SEARCHING_FOR_ROOT_ELEMENT});
+      dispatch({type: START_PAGEOBJECT_INSPECTING});
+      const selectedPOs = [];
+      for (let po of treeData) {
+        if (!po.root || !po.selector) {
+          continue;
+        }
+        const [type, selector] = Object.entries(po.selector)[0];
+        // search root element
+        const callAction = callClientMethod({strategy: strategyMap[type], selector, fetchArray: true});
+        let {elements} = await callAction(dispatch, getState);
+        if (elements.length > 0) {
+          selectedPOs.push(po.name);
+        }
       }
+      dispatch({type: SEARCHING_FOR_ROOT_ELEMENT_COMPLETED, elements: selectedPOs});
```

Comment:
SEARCHING_FOR_ROOT_ELEMENT_COMPLETED to SEARCHING_FOR_CURRENT_PO_COMPLETED

Reply:
updated